runner/host/dmesg.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package host
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/hashicorp/hcdiag/op"
+	"github.com/hashicorp/hcdiag/redact"
+
+	"github.com/hashicorp/hcdiag/runner"
+)
+
+var _ runner.Runner = DMesg{}
+
+// DMesgConfig takes each parameter to configure an DMesg runner implementation.
+type DMesgConfig struct {
+	OS         string           `json:"os"`
+	Timeout    runner.Timeout   `json:"timeout"`
+	Redactions []*redact.Redact `json:"redactions"`
+}
+
+// DMesg accepts
```

Comment:
I forgot to comment earlier, but it looks like this comment is missing some detail.